grobid-core/src/main/java/org/grobid/core/engines/DateParser.java

```diff
@@ -288,16 +308,59 @@ public void normalize(Date date) {
 
                 if (dayPart != -1 && monthPart != -1) {
                     if (dayPart > 0 && dayPart < 32 && monthPart > 0 && monthPart < 13) {
-                        date.setDay(dayPart);
-                        date.setDayString(theDayString);
-                        date.setMonth(monthPart);
-                        date.setMonthString(theMonthString);
-                        date.setYear(yearPart);
+                        normalizedDate.setDay(dayPart);
+                        normalizedDate.setDayString(theDayString);
+                        normalizedDate.setMonth(monthPart);
+                        normalizedDate.setMonthString(theMonthString);
+                        normalizedDate.setYear(yearPart);
                     }
                 }
             }
         }
+        
+        if(validate) {
+            return postValidate(normalizedDate);
+        } else {
+            return normalizedDate;
+        }         
+    }
+
+    /**
+     * Simple and loose date validation, checking: 
+     *  - the year has not more than 4 digits
+     *  - the month and day has not more than 2 digits 
+     *  
+     *  Assuming that incomplete dates of any form and nature can pass by here, only the information that are "out of bounds" will be reverted.
+     *  
+     * @return the date where invalid information are removed or reverted
+     */
+    public static Date postValidate(Date originalDate) {
```

Comment:
minor: I guess in some sense it feels more like "cleaning" than validating. Although loose terms anyway.